converters/converter.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import print_function, unicode_literals
+import os
+import tempfile
+import logging
+from logging import Logger
+from aws_tools.s3_handler import S3Handler
+from shutil import rmtree
+from general_tools.url_utils import download_file
+from general_tools.file_utils import unzip, add_contents_to_zip
+
+
+class Converter(object):
+
+    def __init__(self, logger, s3_handler_class=S3Handler):
+        """
+        :param Logger logger:
+        :param class s3_handler_class:
+        """
+        self.logger = logger
+        self.s3_handler_class = s3_handler_class
+
+        self.convert_log = logging.getLogger('convert_log')
+        self.convert_log.setLevel(logging.INFO)
+        logging.raiseExceptions = 1
+
+        self.data = {}
+        self.options = {}
+        self.job = None
+        self.source = None
+        self.resource = None
+        self.cdn_bucket = None
+        self.cdn_file = None
+
+        self.download_dir = tempfile.mkdtemp(prefix='download_')
+        self.files_dir = tempfile.mkdtemp(prefix='files_')
+        self.input_zip_file = None  # If set, won't download the repo archive. Used for testing
+        self.output_dir = tempfile.mkdtemp(prefix='output_')
+        self.output_zip_file = tempfile.mktemp('.zip')
+
+    def close(self):
```

Comment:
I'd like to see this tested. I think normally `def __exit__(self, exc_type, exc_val, exc_tb):` would be used for this.